models/DataObject/OwnerAwareFieldInterface.php

```diff
@@ -21,8 +21,35 @@ interface OwnerAwareFieldInterface
 {
     /**
      * @param mixed $owner
-     * @param string $fieldname
+     *
+     * @return $this;
+     */
+    public function __setOwner($owner);
```

Comment:
Probably the "double underscore" should mark the methods as "private", but it would be better to use another prefix for this (e.g. only one underscore) because functions/methods that start with `__` are considered as "magic" by PHP and should not be defined in userland as stated here: https://www.php.net/manual/en/language.oop5.magic.php

Reply:
The double underscore is mean to avoid name clashes with existing methods such as "setFieldname". Of course we can change this to something else. Any preferences?

Reply:
The original inspiration was this method I guess which is not magic as well:  "__setDataVersionTimestamp"
I think it would be sufficient to use a single underscore which maybe would be even more consistent as the internally it is stored in _owner, _fieldname and _language.

Reply:
I'd vote for a single underscore, too.

Reply:
done :-)